src/lib/rank_utils.ts

```diff
@@ -22,9 +22,74 @@ interface IRankInfo {
     label: string;
 }
 
-export const MaxRank: number = 36;
+class Rating {
+    unset:boolean;
+    rating:number;
+    deviation:number;
+    volatility:number;
+    rank:number;
+    rank_label:string;
+    partial_rank:number;
+    partial_rank_label:string;
+    rank_deviation_labels:Array<string>;
+    rank_deviation:number;
+}
+
+export const MaxRank: number = 39;
+
+
+const MIN_RATING = 100;
+const MAX_RATING = 6000;
+
+export function rank_to_rating(rank:number) {
+    return 900 * Math.exp(0.032 * rank);
+}
+
+export function rating_to_rank(rating:number) {
+    return Math.log(Math.min(MAX_RATING, Math.max(MIN_RATING, rating)) / 900.0) / 0.032;
+}
+
+export function get_handicap_adjustment(rating:number, handicap:number):number {
+    return rank_to_rating(rating_to_rank(rating) + handicap) - rating;
```

Comment:
Ok I've been thinking and looking over these numbers a lot, several observations and some more information to incorporate into our thoughts.

I agree that `H-0.5` effectively tells Glicko-2 to target something other than 50%, call it 44.5% or whatever. But that's not what we want to do, we want to target 50%, or at least as close as we can get, that's the point of a handicap right? If we target a different value, we're going a compounding effect on relative strength with increased handicaps, it needs to be as close to 50% as possible to make anything more than a couple of stones difference remotely fair / fun.

OK so why aren't we seeing 50%? Seems like the reason would be one or more of the following causes:
 1. I'm wrong and we do need `H-0.5`
 2. The model isn't as great as we would hope
 3. We have bias data.

The first observation I have is that the numbers I've posted incorporate a fair amount of noise from games between players whose' ranks haven't been established yet, and the second thing is all the data I've posted has included 9x9 and 13x13 results as well, the handicap systems for which are a bit more complicated and potentially more bias - that's a problem for another day to solve though. So, I looked at all the data again filtering out all games not 19x19 with a komi of 6.5, and in the case of Glicko, only tallied up games between player's who both had a RD <= 120, so were becoming fairly reliable. I also added the 0 handicap column in the following data, ie even games.


Existing EGF Elo system:
```
     	        0                1                2                3                4                5                6                7                8                9        
20k  	     48.9 [ 3060]     38.2 [  704]     37.8 [  619]     35.0 [  563]     33.4 [  479]     34.9 [  393]     29.9 [  345]     25.1 [  311]     22.9 [  358]     28.5 [  253]
19k  	     48.0 [ 4003]     42.1 [  872]     43.0 [  872]     39.9 [  667]     38.2 [  599]     34.3 [  466]     29.6 [  450]     29.6 [  436]     31.7 [  347]     34.2 [  295]
18k  	     48.2 [ 5641]     44.1 [ 1377]     43.1 [ 1134]     38.8 [  964]     35.4 [  762]     30.5 [  646]     30.4 [  549]     35.4 [  528]     27.8 [  352]     30.4 [  214]
17k  	     48.8 [ 8091]     42.0 [ 1803]     38.4 [ 1647]     38.2 [ 1350]     37.7 [ 1078]     36.4 [  893]     36.7 [  822]     39.1 [  594]     42.0 [  345]     28.6 [  262]
16k  	     48.2 [ 8843]     44.6 [ 2307]     40.8 [ 2101]     38.7 [ 1619]     41.8 [ 1361]     39.6 [ 1119]     42.2 [  816]     41.7 [  509]     35.9 [  329]     36.7 [  245]
15k  	     48.0 [10066]     43.7 [ 3061]     42.7 [ 2459]     43.3 [ 2157]     40.8 [ 1756]     42.2 [ 1205]     43.2 [  708]     37.4 [  473]     31.0 [  352]     44.2 [  197]
14k  	     48.4 [10881]     43.8 [ 3692]     42.8 [ 3354]     42.9 [ 2778]     43.7 [ 1906]     43.1 [ 1098]     40.1 [  673]     39.2 [  467]     40.2 [  301]     39.5 [  210]
13k  	     48.6 [11744]     44.5 [ 4913]     43.2 [ 4246]     45.0 [ 2846]     45.4 [ 1618]     41.6 [  969]     40.4 [  658]     40.9 [  418]     40.9 [  308]     36.0 [  200]
12k  	     47.9 [12812]     45.5 [ 6109]     46.0 [ 4768]     45.3 [ 2860]     48.0 [ 1649]     45.1 [  985]     47.1 [  612]     45.2 [  436]     43.0 [  302]     47.9 [  242]
11k  	     48.2 [14239]     44.6 [ 6826]     45.2 [ 4396]     45.5 [ 2493]     46.1 [ 1442]     46.4 [  890]     46.8 [  587]     42.1 [  404]     44.6 [  307]     45.7 [  188]
10k  	     49.1 [16814]     46.3 [ 6869]     46.5 [ 4158]     45.0 [ 2279]     45.9 [ 1356]     45.0 [  832]     44.8 [  538]     46.2 [  405]     46.0 [  226]     43.9 [  198]
9k   	     48.1 [16724]     45.7 [ 5952]     44.7 [ 3357]     44.9 [ 1938]     45.2 [ 1149]     43.8 [  568]     43.0 [  393]     42.0 [  269]     43.8 [  260]     42.0 [  231]
8k   	     49.0 [15392]     45.4 [ 4866]     46.0 [ 3086]     46.2 [ 1688]     46.9 [  884]     49.0 [  512]     45.9 [  351]     42.2 [  270]     39.6 [  288]     45.4 [  141]
7k   	     48.9 [15737]     46.4 [ 4855]     46.5 [ 2939]     43.5 [ 1448]     45.0 [  743]     47.2 [  407]     42.6 [  326]     43.9 [  326]     47.7 [  172]     51.0 [   49]
6k   	     48.7 [18050]     45.5 [ 4833]     46.8 [ 2339]     43.7 [ 1292]     44.9 [  682]     39.8 [  450]     38.7 [  489]     38.2 [  228]     49.4 [   83]     46.2 [   39]
5k   	     48.2 [19130]     45.8 [ 3853]     44.8 [ 2010]     48.1 [ 1001]     39.9 [  661]     44.7 [  517]     49.2 [  266]     50.5 [  220]     48.4 [   95]     31.2 [   64]
4k   	     48.1 [17566]     46.4 [ 3466]     44.6 [ 1679]     40.3 [  975]     41.8 [  720]     43.7 [  382]     47.5 [  162]     66.7 [  108]     50.0 [   46]     42.0 [   69]
3k   	     48.0 [17370]     41.6 [ 2744]     37.8 [ 1570]     38.9 [ 1092]     40.1 [  526]     45.7 [  247]     50.8 [  185]     36.0 [   86]     56.1 [   98]     50.0 [    4]
2k   	     46.7 [13499]     41.9 [ 1886]     41.3 [ 1148]     40.4 [  591]     42.8 [  421]     43.3 [  275]     36.2 [   58]     54.0 [   87]     50.0 [    4]      0.0 [    0]
1k   	     46.7 [ 7960]     42.0 [ 1458]     43.8 [  468]     31.5 [  432]     40.5 [  264]     46.9 [   64]     53.9 [   89]     66.7 [    6]      0.0 [    0]      0.0 [    0]
1d   	     47.2 [ 5759]     41.6 [  803]     46.0 [  554]     42.3 [  208]     48.6 [  146]     73.1 [   67]    100.0 [    1]      0.0 [    0]      0.0 [    0]      0.0 [    0]
2d   	     43.7 [ 1662]     42.5 [  402]     39.1 [  215]     53.8 [  106]     49.0 [  104]     50.0 [   12]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]
3d   	     45.7 [  488]     35.1 [  259]     46.6 [   88]     56.1 [  155]    100.0 [    3]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]
4d   	     43.8 [  539]     44.0 [  150]     75.3 [   77]     80.0 [    5]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]
5d   	     56.3 [  231]     77.3 [  132]     33.3 [   48]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]

  22k-  19k	     46.0 [10130]     36.9 [ 2308]     32.2 [ 1654]     27.5 [ 1452]     31.6 [  548]
  18k-  15k	     46.4 [48530]     39.9 [11047]     38.4 [ 6758]     36.7 [ 3482]     34.5 [  918]
  14k-  11k	     46.7 [87980]     45.0 [17592]     43.8 [ 6472]     42.0 [ 2943]     42.5 [  840]
  10k-   7k	    47.8 [100749]     45.2 [11485]     45.2 [ 3927]     43.8 [ 2216]     44.1 [  619]
   6k-   3k	     47.4 [94610]     42.4 [ 6949]     43.8 [ 2698]     48.3 [  964]     39.2 [  176]
   2k-   2d	     45.8 [35814]     40.9 [ 2272]     48.4 [  566]     54.6 [   97]      0.0 [    0]
   3d-   6d	     47.6 [ 2134]     57.7 [  163]      0.0 [    0]      0.0 [    0]      0.0 [    0]

Overall: 46.2%
```

Glicko-2 with `H`
```
     	        0                1                2                3                4                5                6                7                8                9        
20k  	     44.4 [ 1833]     46.7 [  184]     43.4 [  136]     38.8 [   98]     45.6 [   79]     48.1 [   52]     25.0 [   32]     25.0 [   20]     31.2 [   16]     24.1 [   29]
19k  	     44.7 [ 2620]     45.2 [  239]     44.4 [  225]     36.6 [  164]     37.9 [  103]     42.2 [   64]     37.5 [   40]     29.7 [   37]     33.3 [   21]     28.6 [   21]
18k  	     44.7 [ 3255]     48.8 [  383]     46.4 [  323]     43.9 [  198]     40.0 [  145]     44.0 [  100]     31.1 [   61]     28.6 [   42]     44.8 [   29]     40.8 [   49]
17k  	     48.1 [ 4010]     44.8 [  495]     44.7 [  385]     42.9 [  268]     39.4 [  175]     47.2 [  125]     36.7 [   79]     40.7 [   54]     33.3 [   30]     42.9 [   35]
16k  	     47.3 [ 4830]     46.8 [  633]     49.5 [  501]     45.6 [  351]     43.2 [  250]     48.8 [  164]     41.3 [   75]     33.3 [   48]     46.6 [   58]     50.0 [   42]
15k  	     47.3 [ 5644]     48.4 [  821]     45.0 [  658]     44.4 [  477]     46.7 [  285]     44.8 [  183]     44.1 [  127]     43.5 [   85]     38.0 [   50]     35.0 [   40]
14k  	     48.9 [ 5991]     45.5 [ 1074]     43.1 [  814]     47.1 [  520]     48.7 [  339]     44.7 [  197]     52.4 [  143]     32.9 [   79]     51.1 [   47]     34.1 [   41]
13k  	     47.9 [ 6067]     46.0 [ 1251]     46.7 [  922]     47.3 [  573]     46.6 [  354]     48.1 [  189]     35.7 [  115]     44.4 [   72]     25.5 [   51]     37.8 [   37]
12k  	     48.6 [ 6907]     47.3 [ 1549]     45.9 [ 1086]     43.8 [  598]     46.5 [  359]     47.8 [  205]     46.4 [   97]     33.3 [   57]     45.2 [   42]     33.3 [   30]
11k  	     48.9 [ 7929]     47.0 [ 1809]     46.1 [ 1102]     47.2 [  652]     49.5 [  325]     41.5 [  176]     44.0 [  109]     43.4 [   53]     48.7 [   39]     41.9 [   31]
10k  	     49.2 [ 8385]     47.4 [ 1807]     43.3 [ 1078]     47.4 [  582]     42.0 [  288]     39.5 [  162]     47.5 [   80]     42.9 [   63]     35.4 [   48]     56.0 [   25]
9k   	     47.6 [ 8733]     45.3 [ 1673]     47.8 [  967]     45.8 [  542]     41.4 [  251]     44.3 [  140]     39.3 [   84]     51.0 [   49]     26.3 [   38]     42.4 [   33]
8k   	     48.3 [ 9841]     46.7 [ 1599]     49.2 [  884]     44.4 [  399]     40.8 [  255]     43.4 [  122]     50.5 [   97]     39.2 [   51]     54.5 [   44]     40.7 [   27]
7k   	     48.8 [10899]     46.0 [ 1544]     47.1 [  731]     41.7 [  367]     41.7 [  187]     43.8 [  105]     37.5 [   64]     48.8 [   43]     42.9 [   35]     77.3 [   22]
6k   	     47.8 [11379]     45.6 [ 1253]     42.5 [  617]     44.3 [  289]     42.8 [  152]     44.0 [   84]     45.8 [   72]     44.4 [   63]     53.8 [   26]     27.3 [   11]
5k   	     48.6 [10812]     43.1 [ 1006]     43.3 [  506]     40.7 [  199]     53.2 [  124]     42.9 [   84]     46.0 [   50]     43.8 [   48]     53.8 [   13]     50.0 [    8]
4k   	     48.0 [ 9766]     45.4 [  872]     46.1 [  393]     46.3 [  203]     47.1 [  153]     50.5 [   97]     45.7 [   46]     63.0 [   27]     33.3 [    9]     16.7 [   12]
3k   	     48.8 [ 7871]     46.7 [  698]     42.2 [  313]     40.4 [  223]     38.5 [  135]     44.6 [   92]     45.2 [   31]     52.6 [   19]     16.7 [    6]     43.8 [   16]
2k   	     48.6 [ 5794]     39.8 [  512]     43.3 [  321]     45.4 [  183]     51.2 [   84]     39.7 [   78]     54.2 [   24]     46.7 [   15]     31.6 [   19]     21.7 [   23]
1k   	     47.6 [ 4165]     45.5 [  413]     49.1 [  173]     42.9 [   91]     40.0 [   95]     41.8 [   55]     33.3 [   18]     38.5 [   13]     22.2 [    9]     50.0 [    2]
1d   	     48.9 [ 2650]     47.6 [  246]     45.3 [  106]     47.8 [   90]     45.7 [   46]     53.8 [   13]     46.7 [   15]     38.5 [   13]      0.0 [    2]      0.0 [    1]
2d   	     48.0 [ 1394]     40.4 [  146]     48.0 [   75]     38.9 [   36]     42.1 [   19]     33.3 [   12]     16.7 [   12]    100.0 [    1]     50.0 [    2]      0.0 [    0]
3d   	     49.8 [  607]     47.7 [   88]     49.1 [   55]     42.9 [   14]     53.3 [   15]     53.8 [   13]     66.7 [    3]      0.0 [    0]      0.0 [    0]      0.0 [    0]
4d   	     50.2 [  237]     40.6 [   64]     56.2 [   16]     44.4 [   18]     50.0 [   16]     33.3 [    6]      0.0 [    1]      0.0 [    0]      0.0 [    0]      0.0 [    0]
5d   	     53.5 [  159]     37.5 [   40]     66.7 [   15]     61.5 [   26]     25.0 [    4]     50.0 [    2]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]

  22k-  19k	     44.8 [ 7717]     39.5 [  684]     37.0 [  300]     29.3 [  147]     23.4 [   77]
  18k-  15k	     47.0 [21938]     43.8 [ 2149]     43.8 [  914]     39.4 [  396]     42.2 [  166]
  14k-  11k	     48.0 [36501]     46.9 [ 3720]     45.4 [ 1231]     39.8 [  440]     36.7 [  139]
  10k-   7k	     48.0 [48141]     43.9 [ 2871]     43.2 [  854]     42.9 [  371]     52.3 [  107]
   6k-   3k	     47.8 [45486]     43.8 [ 1478]     45.7 [  556]     47.9 [  211]     34.0 [   47]
   2k-   2d	     47.7 [15995]     44.9 [  644]     41.0 [  227]     36.5 [   74]     23.1 [   26]
   3d-   6d	     50.4 [ 1458]     48.5 [   97]     48.0 [   25]      0.0 [    0]      0.0 [    0]

Overall: 47.3%
```

Glicko-2 with `H-5`
```
     	        0                1                2                3                4                5                6                7                8                9        
20k  	     44.9 [ 1834]     46.9 [  177]     40.2 [  107]     37.5 [   96]     37.3 [   83]     50.0 [   56]     28.6 [   42]     44.8 [   29]     12.5 [   16]     30.8 [   26]
19k  	     44.3 [ 2559]     43.7 [  229]     45.5 [  220]     41.7 [  163]     40.3 [  119]     43.7 [   71]     31.7 [   60]     30.8 [   39]     40.0 [   35]     32.0 [   25]
18k  	     44.1 [ 3287]     49.9 [  373]     44.7 [  320]     40.1 [  227]     41.8 [  184]     32.7 [  110]     28.0 [   75]     30.0 [   60]     39.4 [   33]     48.8 [   41]
17k  	     47.8 [ 4035]     44.4 [  502]     44.3 [  409]     41.2 [  289]     35.8 [  201]     43.4 [  145]     32.6 [   86]     40.8 [   71]     26.1 [   46]     40.0 [   45]
16k  	     47.2 [ 4868]     49.9 [  686]     45.5 [  521]     44.7 [  365]     40.7 [  248]     46.1 [  191]     43.6 [   94]     34.3 [   67]     40.0 [   60]     43.8 [   48]
15k  	     47.6 [ 5800]     47.1 [  849]     44.9 [  699]     47.2 [  511]     48.6 [  325]     45.1 [  204]     42.0 [  138]     35.7 [   98]     39.3 [   56]     25.6 [   39]
14k  	     49.0 [ 6198]     45.1 [ 1132]     42.8 [  894]     43.8 [  589]     45.0 [  371]     43.6 [  236]     43.0 [  151]     34.5 [   87]     40.8 [   49]     30.4 [   46]
13k  	     47.7 [ 6363]     45.4 [ 1356]     46.3 [  984]     45.1 [  652]     47.7 [  392]     50.0 [  228]     37.6 [  141]     45.5 [   77]     26.8 [   56]     34.8 [   46]
12k  	     48.7 [ 7283]     46.6 [ 1644]     45.3 [ 1171]     45.9 [  656]     47.6 [  391]     43.7 [  215]     56.4 [  110]     33.3 [   75]     42.0 [   50]     46.2 [   39]
11k  	     48.6 [ 8257]     46.0 [ 1915]     44.7 [ 1222]     44.4 [  701]     47.0 [  345]     42.5 [  200]     39.7 [  116]     36.7 [   60]     40.0 [   55]     37.0 [   46]
10k  	     48.8 [ 8900]     46.8 [ 1941]     46.4 [ 1193]     47.4 [  618]     40.3 [  293]     43.2 [  169]     45.6 [   90]     40.0 [   55]     31.5 [   54]     44.8 [   29]
9k   	     47.4 [ 9153]     44.2 [ 1772]     47.4 [  997]     44.0 [  570]     38.9 [  280]     44.5 [  146]     38.6 [  101]     55.8 [   43]     33.3 [   48]     37.9 [   58]
8k   	     48.9 [10330]     47.4 [ 1669]     47.5 [  912]     44.4 [  412]     44.8 [  259]     47.6 [  124]     49.1 [  110]     37.9 [   66]     41.5 [   41]     31.4 [   35]
7k   	     47.9 [11377]     45.3 [ 1603]     46.2 [  754]     38.9 [  386]     37.8 [  193]     38.1 [  105]     36.2 [   80]     52.9 [   51]     37.5 [   40]     55.6 [   27]
6k   	     48.1 [11678]     45.8 [ 1269]     40.7 [  659]     42.4 [  290]     44.8 [  154]     39.8 [   93]     51.3 [   78]     40.0 [   65]     42.9 [   35]     30.0 [   10]
5k   	     48.2 [11018]     42.8 [ 1053]     45.2 [  502]     41.8 [  232]     49.3 [  148]     54.7 [  106]     41.1 [   56]     50.0 [   42]     83.3 [    6]     21.1 [   19]
4k   	     48.0 [ 9685]     45.2 [  877]     43.4 [  431]     45.5 [  222]     36.4 [  162]     43.5 [  108]     65.3 [   49]     50.0 [   28]     40.0 [   10]     28.6 [   14]
3k   	     48.8 [ 7617]     43.9 [  660]     41.9 [  346]     37.0 [  246]     41.5 [  130]     45.5 [  101]     27.6 [   29]     45.5 [   22]      4.3 [   23]     33.3 [   30]
2k   	     48.2 [ 5563]     40.5 [  484]     43.6 [  307]     43.5 [  154]     42.5 [   87]     36.1 [   72]     44.4 [   27]     25.0 [   20]     38.1 [   21]     27.3 [   11]
1k   	     47.2 [ 3977]     46.4 [  399]     51.7 [  147]     45.9 [   98]     40.0 [   85]     43.8 [   32]     26.3 [   19]     38.5 [   13]     23.1 [   13]      0.0 [    1]
1d   	     49.8 [ 2348]     42.5 [  228]     35.4 [   99]     41.6 [   77]     46.3 [   41]     44.4 [    9]     40.0 [   20]     46.2 [   13]      0.0 [    0]      0.0 [    0]
2d   	     48.5 [ 1149]     37.8 [  119]     44.6 [   65]     43.8 [   32]     45.0 [   20]     30.3 [   33]     33.3 [    3]      0.0 [    0]    100.0 [    1]      0.0 [    0]
3d   	     48.2 [  415]     41.2 [   85]     50.0 [   36]     40.0 [   15]     44.4 [   18]     42.9 [    7]    100.0 [    1]      0.0 [    0]      0.0 [    0]      0.0 [    0]
4d   	     52.1 [  240]     41.1 [   56]     41.7 [   12]     58.3 [   24]     33.3 [   12]     40.0 [    5]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]
5d   	     52.4 [  124]     48.8 [   43]     57.9 [   19]     66.7 [   12]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]

  22k-  19k	     44.9 [ 7509]     39.9 [  682]     37.1 [  340]     33.5 [  185]     29.3 [   75]
  18k-  15k	     46.8 [22349]     43.5 [ 2350]     40.9 [ 1043]     35.8 [  491]     39.9 [  173]
  14k-  11k	     47.7 [38419]     45.5 [ 4097]     44.5 [ 1397]     37.3 [  509]     36.7 [  177]
  10k-   7k	     47.8 [50601]     42.9 [ 3011]     43.2 [  925]     41.0 [  398]     40.9 [  149]
   6k-   3k	     47.7 [45795]     42.1 [ 1584]     46.9 [  620]     41.6 [  231]     28.8 [   73]
   2k-   2d	     47.6 [14885]     43.3 [  594]     37.2 [  215]     34.6 [   81]     25.0 [   12]
   3d-   6d	     49.6 [ 1197]     49.4 [   81]     46.2 [   13]      0.0 [    0]      0.0 [    0]

Overall: 47.0%
```

Looking at these numbers we note that it's reasonably close to 50% for H0, but every system, egf or glicko seems to have a bias that's against what we would expect from http://senseis.xmp.net/?Komi%2FStatistics , we should, if anything, be seeing a slight bias towards black.

Which led me to wonder if we had some bias data, which it turns out we do. We made the decision years ago that when deciding who get's white, the stronger player takes white. We made this decision because it's tradition to do this when the ranks aren't equal, and since there's a natural bias against white, we figured games would on average be more fun if this was extrapolated to a rating difference as well. I think that decision might play a minor role in a rank compression problem, but it doesn't seem to be particularly harmful overall to either the egf or glicko system. (Still, going forward, I think I'm going to change that to be random when the rank difference is 0.). Regardless, I believe that accounts for the bias against black that we're seeing across all the systems, and it also means we shouldn't be trying to compensate for that bias as that's a bias coming from a difference in skill, not a bias in the relative difficulty in the actual game, so we should still target 50%.

Going back to `H` vs `H-0.5`, with handicaps there does indeed seem to be a bias towards white, but I still say *there shouldn't be*, the goal by definition is to have a fair game between two players if you use a handicap that is the difference in your ranks, the goal is no bias, so this to me indicates that there is a problem with the model. The percentages are roughly comparable between the Glicko-2 and EGF systems, which makes sense since we derived the fit from EGF system, and since the EGF system has this `H-0.5`, that could very well account for the problem with the model, and since we used ranks derived from that system to come up with a fit for Glicko-2 to Rank, it makes sense that this bias is still present in our model. So this exercise is telling me maybe we shouldn't do that. I think it's telling me that the EGF ranks need to be expanded a bit so that on average a 1 rank difference in strength translates to less difference in skill, so the outcomes go back to what they should be, which is ideally (by definition) 50%.